File: frontend/src/types/projectMember.ts
import { RoleType } from "./enums";

export interface ProjectMemberCreate {
  user_id: string;
  role: RoleType;
}

export interface ProjectMemberUpdate {
  role: RoleType;
}

export interface ProjectMemberResponse {
  id: string;
  user_id: string;
  project_id: string;
  role: RoleType;
  created_at: string;
}

File: backend/app/schemas/project_member.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.enums import Role


class ProjectMemberCreate(BaseModel):
    user_id: UUID
    role: Role = Role.USER


class ProjectMemberUpdate(BaseModel):
    role: Role


class ProjectMemberResponse(BaseModel):
    id: UUID
    user_id: UUID
    project_id: UUID
    role: Role
    created_at: datetime
